### PR TITLE
Choose pipe number on initialize

### LINKED ISSUE #203: Ability to switch between open clients of Discord
So I've been using the [Discord Integration](https://plugins.jetbrains.com/plugin/10233-discord-integration) for IntelliJ which uses the Discord rpc but I experienced a problem with which cannot be fixed by the plugin author himself.

At the moment I'm simultaneously running Discord (alt account) and Discord PTB (main account) and the rpc will always connect to the main Discord window. The problem I have with this is that I want the rich presence to be displayed on my main account (the PTB).

With that in mind, I'd like it if the rpc would detect if there are multiple clients open (e.g. normal, ptb, canary) and add the ability to chose which to send the presence to if not multiple of them.

### PR DESCRIPTION
**Pull request by msciotti**
Fixes #203

### CODE REVIEW

File: examples/button-clicker/Assets/DiscordRpc.cs [hunk 1]
```diff
@@ -5,194 +5,199 @@
 
 public class DiscordRpc
 {
-    [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
```
Comment (Bluexin) [inline examples/button-clicker/Assets/DiscordRpc.cs]: Is the switch to 2-space indentation on purpose?
I'm not very familiar with C# but I believe official convention is to use 4 spaces.

Comment (msciotti) [reply examples/button-clicker/Assets/DiscordRpc.cs]: I think it's just my editor being annoying. Probably don't have the formatter set up correctly. I can fix before merging!

**Comment by Bluexin** [comment]
Do you think it would be possible to add a "discovery" thing in the base library? To be able to list available clients for example (with their corresponding pipe number).

**Comment by msciotti** [comment]
Possible yeah, although there's not a lot of magic to it. The first client you open will be at 0, the second client at 1, etc. etc. It probably wouldn't be too hard to do that, but I'd PR that separately I think.

**Comment by msciotti** [comment]
Still need to finish this, sorry for the delay. Rebase has been a bit of a cluster.

**Comment by janisozaur** [comment]
This added parameter changes API and ABI, while it looks to be defaulted on some wrappers. Is there a reason why the C API doesn't default this parameter?

**Comment by IntelOrca** [comment]
This breaks consumers as you have changed the public API. Would you be able to add a new version, e.g. `Discord_Initialize_2` or `Discord_Initialize_Pipe`?

It would also be better to pass a struct instead of individual parameters for new APIs as that allows you to more easily extend them without breaking compatibility.

**Comment by msciotti** [comment]
Forgive my ignorance, how does this break things for existing users? None of this was changed in Discord itself, just within the native SDK stuff. If you're still working on an old release version, there should be no issues with already-existing implementations?

**Comment by janisozaur** [comment]
You broke ABI and API so clients using older `discord-rpc` no longer work until updated. The change is trivial, so it's not a problem of complexity, but rather why the need to break compatibility in the first place?

**Comment by IntelOrca** [comment]
@msciotti This impacts Linux users more as their ecosystem will be more likely that libraries are updated separately from the applications. So any minor update to a library can potentially break all applications that use it.

**Comment by msciotti** [comment]
Sorry, please forgive my ignorance here. The Discord application itself has not changed. This is a breaking change, but only if you actually update your shipped application to whatever is currently on master. This does not affect games or applications that are currently live and have not been updated.

I'm happy to revert the commit, which I guess I'll do anyway, and re-attempt this PR when I have time to do it again.

**Comment by IntelOrca** [comment]
@msciotti thanks, the problem is that some people build our game with the latest master of discord and of course the build started to fail for them due to this commit.

**Pull request merged**